test/test_quantized.py

```diff
@@ -273,6 +273,49 @@ def test_cat(self, X, num, axis, relu):
             cat_q = q_cat_op(tensors_q, axis=axis, scale=scale,
                              zero_point=zero_point)
 
+
+@unittest.skipIf(
+    TEST_WITH_UBSAN or not torch.fbgemm_is_cpu_supported(),
+    " Quantized Linear requires FBGEMM. FBGEMM does not play"
+    " well with UBSAN at the moment, so we skip the test if"
+    " we are in a UBSAN environment.",
+)
+class TestDynamicQuantizedLinear(unittest.TestCase):
+    """Tests the correctness of the dynamic quantized linear and linear_relu op."""
+    @given(
+        use_bias=st.booleans(),
+        use_relu=st.booleans(),
+    )
+    def test_qlinear(self, use_bias, use_relu):
+        batch_size = 1
+        input_channels = 2
+        output_channels = 2
+
+        qlinear_prepack = torch.ops.quantized.fbgemm_linear_prepack
+        if use_relu:
+            qlinear_dynamic = torch.ops.quantized.fbgemm_linear_relu_dynamic
+        else:
+            qlinear_dynamic = torch.ops.quantized.fbgemm_linear_dynamic
+
+        W_fp32 = torch.tensor([[-150, 100], [100, -150]], dtype=torch.float)
+        b_fp32 = torch.tensor([13, -20], dtype=torch.float) if use_bias else None
+        X_fp32 = torch.tensor([[100, -150]], dtype=torch.float)
+
+        W_scale, W_zp = _calculate_dynamic_qparams(W_fp32, torch.qint8)
+        W_q = torch.quantize_linear(W_fp32, scale=W_scale, zero_point=W_zp, dtype=torch.qint8)
+
+        # Weight prepacking operator for dynamic quantized Linear
+        W_prepack = qlinear_prepack(W_q)
+        # Dynamic quantized Linear operator with prepacked weight
+        Y_fp32 = qlinear_dynamic(X_fp32, W_prepack, b_fp32)
+
+        Y_fp32_ref = F.linear(X_fp32, W_fp32, b_fp32)
+        if use_relu:
+            Y_fp32_ref[Y_fp32_ref < 0.0] = 0.0
+
+        torch.testing.assert_allclose(Y_fp32, Y_fp32_ref, rtol=0.0001, atol=1e-3)
```

Comment:
Can we have  a test that is exact: If you quantize and dequantize the activations prior to F.linear, can you get an exact match?

Reply:
Right: currently we use a special example (the same as https://github.com/pytorch/pytorch/blob/master/test/test_jit.py#L11177-L11198), so we can have the exact match.